test/distributed/test_launcher.py

```diff
@@ -51,6 +52,52 @@ def test_launch_user_script(self):
         ]
         launch.main(args)
 
+    def test_launch_deprecated_warning(self):
+        script_path = path("bin/test_script.py")
+        processes = []
+
+        for node_rank in [0, 1]:
+            launch_script = f'''
+import torch
+import torch.distributed.launch as launch
+launch.main([
+    '--nnodes', '2',
+    '--node_rank', '{node_rank}',
+    '{script_path}'])
+'''
+
+            processes.append(subprocess.Popen(
+                ['python', '-c', launch_script],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE))
+
+        for rank, process in enumerate(processes):
+            process_out, process_err = process.communicate()
+            if rank == 0:
+                self.assertIn('launch is deprecated', str(process_err))
+            else:
+                self.assertNotIn('launch is deprecated', str(process_err))
+
+    def test_omp_threads_warning(self):
+        script_path = path("bin/test_script.py")
+        processes = []
+
+        for node_rank in [0, 1]:
+            processes.append(subprocess.Popen([
+                'torchrun',
+                '--nnodes', '2',
+                '--nproc_per_node', '2',
+                '--node_rank', str(node_rank),
+                script_path],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE))
+
+        for rank, process in enumerate(processes):
+            process_out, process_err = process.communicate()
+            if rank == 0:
+                self.assertIn('Setting OMP_NUM_THREADS', str(process_err))
```

Comment:
For some reason, `process_err` is blank if the `test_launch_user_script` test is run before this one. Not sure why yet

Reply:
Ah, it was just because the `OMP_NUM_THREADS` env var was getting set by `test_launch_user_script`, of course